components/merkledb/src/db.rs

```diff
@@ -453,6 +454,15 @@ pub trait Database: Send + Sync + 'static {
     /// will be returned. In case of an error, the method guarantees no changes are applied to
     /// the database.
     fn merge_sync(&self, patch: Patch) -> Result<()>;
+
+    /// Creates checkpoint of this database in the given directory.
```

Comment:
Sorry for not asking previously: do we _really_ need this method in the `Database` trait, or would implementing it for a specific implementation (`RocksDB`) suffice?

Reply:
> do we really need this method in the Database trait

Actually no. However it was hard to find a clean way to downcast `dyn Database` to `&RocksDb`. I'll need to add additional dependency on [`downcast-rs`][1] or to reimplement the same functionality.

[1]: https://crates.io/crates/downcast-rs

Reply:
Please look into the new version of this PR.

Reply:
Can you describe in more detail how you plan to use `RocksDB` for backup? It seems strange to me that the current approach assumes we've just got a pointer to `dyn Database` and need to downcast it. Where in reality, the workflow may look something like this:

```rust
fn create_blockchain(db: Arc<dyn Database>) {
    // This emulates `Blockchain::new(..)`
}

fn backup(db: Arc<RocksDB>) {
    // ...
}

// We need to initialize a concrete DB *somewhere*.
let db: Arc<RocksDB> = Arc::new(RocksDB::new(..));
// Here, we upcast the `Arc` pointer to use dynamic dispatch.
create_blockchain(Arc::clone(&db) as Arc<dyn Database>);
// But we still know that the original pointer points to `RocksDB`!  
backup(db);
```

Reply:
> Can you describe in more detail how you plan to use RocksDB for backup?

Functionality will be exposed as a Blockchain method and used in service hooks and API handlers.